FindOpenStudioSDK.cmake

```diff
@@ -1,78 +1,148 @@
-    set(OPENSTUDIO_EXT "tar.gz")
-  elseif(WIN32)
-    set(OPENSTUDIO_EXPECTED_HASH f4329ab725fe3e68203a112f8b9655e5)
-    set(OPENSTUDIO_PLATFORM "Windows")
-    set(OPENSTUDIO_EXT "tar.gz")
-  endif()
+  # base link for develop builds. (Using https will fail)
+  set(OPENSTUDIO_BASELINK_CI "http://openstudio-ci-builds.s3-website-us-west-2.amazonaws.com/develop"
+    CACHE STRING "Base link to where the openstudio develop archives are hosted" FORCE)
 
-  set(OPENSTUDIO_ARCHIVE_BASENAME "OpenStudio-${OPENSTUDIO_VERSION}${OPENSTUDIO_VERSION_SHA}-${OPENSTUDIO_PLATFORM}")
-  set(OPENSTUDIO_ARCHIVE_NAME "${OPENSTUDIO_ARCHIVE_BASENAME}.${OPENSTUDIO_EXT}"
-      CACHE STRING "Archive Name, with extension" FORCE)
+  # Make subdir if it doesn't exist
+   execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory "${OPENSTUDIO_ARCHIVE_DIR}/")
 
-  if(EXISTS "${PROJECT_BINARY_DIR}/${OPENSTUDIO_ARCHIVE_NAME}")
-    file(MD5 "${PROJECT_BINARY_DIR}/${OPENSTUDIO_ARCHIVE_NAME}" OPENSTUDIO_HASH)
+  # First pass: if the archive exists, compute the MD5
+  if(EXISTS "${OPENSTUDIO_ARCHIVE_DIR}/${OPENSTUDIO_ARCHIVE_NAME}")
+    file(MD5 "${OPENSTUDIO_ARCHIVE_DIR}/${OPENSTUDIO_ARCHIVE_NAME}" OPENSTUDIO_HASH)
   endif()
-  if(NOT EXISTS "${PROJECT_BINARY_DIR}/${OPENSTUDIO_ARCHIVE_NAME}" OR NOT "${OPENSTUDIO_HASH}" MATCHES "${OPENSTUDIO_EXPECTED_HASH}")
-    set(OPENSTUDIO_URL "${OPENSTUDIO_BASELINK}/${OPENSTUDIO_ARCHIVE_NAME}")
-    string(REPLACE "+" "%2B" OPENSTUDIO_URL ${OPENSTUDIO_URL})
-    message(STATUS "Downloading OpenStudio SDK: ${OPENSTUDIO_URL}")
 
-    file(DOWNLOAD "${OPENSTUDIO_URL}" "${PROJECT_BINARY_DIR}/${OPENSTUDIO_ARCHIVE_NAME}"
-       INACTIVITY_TIMEOUT 300 # 5-min timeout
-       SHOW_PROGRESS
-       EXPECTED_MD5 ${OPENSTUDIO_EXPECTED_HASH})
+  # If the archive doesn't exist, or the archive has the wrong MD5, then resort to download
+  if(NOT EXISTS "${OPENSTUDIO_ARCHIVE_DIR}/${OPENSTUDIO_ARCHIVE_NAME}" OR NOT "${OPENSTUDIO_HASH}" MATCHES "${OPENSTUDIO_EXPECTED_HASH}")
+
+    if(NOT EXISTS "${OPENSTUDIO_ARCHIVE_DIR}/${OPENSTUDIO_ARCHIVE_NAME}")
+      message(STATUS "Archive doesn't exist at \"${OPENSTUDIO_ARCHIVE_DIR}/${OPENSTUDIO_ARCHIVE_NAME}\"")
+    else()
+      message(STATUS
+        "Existing archive md5sum HASH mismatch\n"
+        "     for file: ${OPENSTUDIO_ARCHIVE_DIR}/${OPENSTUDIO_ARCHIVE_NAME}\n"
+        "       expected hash: [${OPENSTUDIO_EXPECTED_HASH}]\n"
+        "         actual hash: [${OPENSTUDIO_HASH}]\n"
+      )
+    endif()
 
     # Remove the old extracted dir if exists
-    execute_process(COMMAND ${CMAKE_COMMAND} -E remove_directory "${PROJECT_BINARY_DIR}/${OPENSTUDIO_ARCHIVE_BASENAME}")
+    execute_process(COMMAND ${CMAKE_COMMAND} -E remove_directory "${OPENSTUDIO_ARCHIVE_DIR}/${OPENSTUDIO_ARCHIVE_BASENAME}")
+
+    # Try with the official releases first, then fall back to the CI (develop) nightly builds
+    foreach(BASELINK IN LISTS OPENSTUDIO_BASELINK_RELEASE OPENSTUDIO_BASELINK_CI)
```

Comment:
Here it is: try multiple URLs until one does the job. Adding a third url for our own builds of OpenStudio is a breeze, just need to add it here.